File: Programmers/Lv1/Lv1_체육복.py
def solution(n, lost, reserve):
    answer = 0
    for i in range(1, n + 1):
        if i not in lost:  # ???껋뼱踰꾨┛ ?숈깮
            answer += 1
        else:
            if i in reserve:  # ?껋뼱踰꾨졇吏留??щ텇???덈뒗 ?숈깮
                answer += 1
                reserve.remove(i)
                lost.remove(i)

    for i in lost:  # ?껋뼱踰꾨━怨??щ텇???놁뼱??鍮뚮젮???섎뒗 ?숈깮
        if i - 1 in reserve:
            answer += 1
            reserve.remove(i - 1)

        elif i + 1 in reserve:
            answer += 1
            reserve.remove(i + 1)
    return answer

print(solution(5,[2,4],[3]))

File: Programmers/Lv1/Lv1_체육복.cpp
#include <string>
#include <vector>
#include <list>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = n - lost.size();
	list<int> l(begin(reserve), end(reserve));

	for (int i = 0; i < lost.size(); i++) {
		auto it = l.begin();
		for (int j = 0; j < l.size(); j++) {
			if (abs(lost[i] - *it) <= 1) {
				answer++;
				l.erase(it);
				break;
			}
			it++;
		}
	}
	return answer;
}
